Check input resolution and exit if it's the same or less than output resolution

If you give a 1280x720 video to track4k with an output resolution of 1280x720, it fails with an assertion error.

There's nothing meaningful to do in this instance (may as well use the input video unchanged), so track4k should check at startup and exit with an error message if input resolution is equal to or smaller than output resolution.

```
OpenCV: FFMPEG: tag 0x34363258/'X264' is not supported with codec id 28 and format 'mp4 / MP4 (MPEG-4 Part 14)'
OpenCV: FFMPEG: fallback to use tag 0x00000021/'!???'
Input frame resolution: Width=1280  Height=720 of nr#: 7499
Input codec type:
Video Duration (Seconds): 299
FPS: 25
OpenCV Error: Assertion failed (0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows) in Mat, file /usr/local/src/track4k/opencv/modules/core/src/matrix.cpp, line 526
terminate called after throwing an instance of 'cv::Exception'
  what():  /usr/local/src/track4k/opencv/modules/core/src/matrix.cpp:526: error: (-215) 0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows in function Mat
```

--- a/source/mainDriver.cpp
+++ b/source/mainDriver.cpp
@@ -96,6 +96,12 @@ int main(int argc, char *argv[]) {
     cout << "\nStage 1 Complete" << endl;
     cout << "----------------------------------------\n" << endl;
 
+    // Check that input resolution is larger than output resolution, otherwise there's nothing to do
+    if ((persistentData.videoDimension.width <= cropWidth) || (persistentData.videoDimension.height < cropHeight)) {
+        cerr << "Input video size is the same or smaller than output size: nothing to do." << endl;
+        return EXIT_FAILURE;
+    }
+
     vector<Rect> r;
     cout << "\n----------------------------------------" << endl;
     cout << "Stage [2 of 3] - Lecturer Tracking" << endl;

